test/functional/shared.js

```diff
@@ -189,6 +189,27 @@ function withMonitoredClient(commands, options, callback) {
   };
 }
 
+/**
+ * Safely perform a test with an arbitrary cursor.
+ *
+ * @param {Function} cursor any cursor that needs to be closed
+ * @param {(cursor: Object, done: Function) => void} body test body
+ * @param {Function} done called after cleanup
+ */
+function withCursor(cursor, body, done) {
```

Comment:
Just a thought, and not one I wanted to block this review on:

This helper doesn't account for tests which return a promise, which is a similar discussion to ones we've had previously in `withClient` and friends. I wonder if what we eventually want is some helper like `withCloseable` which is internally used by `withClient`, `withMonitoredClient`, `withCursor` and maybe even something like `withSession` (though this is an existing helper in the public API). Food for thought

Reply:
Agreed it would be nice for `withMonitoredClient` and `withCursor` to support promises as well as callbacks, the way `withClient` does. I think refactoring to use `withCloseable` would be a nice reduction of duplicated logic but I don't see how it would add much utility. Perhaps it's something we can work into a future PR where we're trying to write promise-based tests that need monitored clients or cursors?